### DISCUSSION TITLE
why use -kn the texture will be disordered?

### DISCUSSION THREAD

**Discussion by 554814925** [zeux/meshoptimizer#500] [Q&A] [answered]
I need to keep node name, but when i use -kn,the texture will be disordered.
this is orginal model :
http://www.3dyzt.com/temp/house.glb

gltfpack.exe -i house.glb -o test.glb -tc -kn

this is correct:
![normal](https://user-images.githubusercontent.com/61701296/197503498-2c242cde-7b7c-479e-81b9-b0ae2a40b56a.png)

this is wrong:
![微信图片_20221024181252](https://user-images.githubusercontent.com/61701296/197503545-f296fee0-841a-4cc6-971f-c608e9f7846f.png)

**Comment by zeux** [comment, marked as answer]
It looks like there's a large amount of differences between scales of different nodes, which confuses the quantization logic and that results in a significant error for some objects. You can use `-vpf` (available in latest builds that you can download from https://github.com/zeux/meshoptimizer/actions/runs/3299430986) to fix that, which will likely become the default setting in the future.
